OptimizelySDK/Entity/FeatureDecision.cs

```diff
@@ -0,0 +1,52 @@
+﻿/* 
+ * Copyright 2017, Optimizely
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+using System;
+using System.Collections.Generic;
+
+namespace OptimizelySDK.Entity
+{
+    public class FeatureDecision : IEquatable<object>
+    {
+        public const string DECISION_SOURCE_EXPERIMENT = "experiment";
+        public const string DECISION_SOURCE_ROLLOUT = "rollout";
+
+        public string ExperimentId { get; set; }
+        public string VariationId { get; set; }
+        public string Source { get; set; }
+
+        public FeatureDecision(string experimentId, string variationId, string source)
+        {
+            ExperimentId = experimentId;
+            VariationId = variationId;
+            Source = source;
+        }
+
+        public override bool Equals(object obj)
+        {
+            var decision = obj as FeatureDecision;
+            return decision != null &&
+                   ExperimentId == decision.ExperimentId &&
+                   VariationId == decision.VariationId &&
+                   Source == decision.Source;
+        }
+
+        public override int GetHashCode()
```

Comment:
not needed.

Reply:
usually it is used in java.

Reply:
Visual Studio generates warning if we override Equals() but not GetHashCode() that's why I implemented it also.

Warning	CS0659	'FeatureDecision' overrides Object.Equals(object o) but does not override Object.GetHashCode()